rstream/client.py

```diff
@@ -461,14 +474,48 @@ def __init__(
         self._clients: dict[Addr, Client] = {}
 
     async def get(self, addr: Optional[Addr] = None) -> Client:
-        if addr is None:
-            addr = self.addr
+        """Get a client according to `addr` parameter
+
+        If class param `load_balancer_mode` is True, we create a connection via the LB
+        in a loop until the desired node is returned.
+
+        If no `addr` is supplied, then it is assumed the exact node is not important
+        and `load_balancer_mode` is ignored.
+        """
+        desired_addr = addr or self.addr
+
+        if desired_addr not in self._clients:
+            if addr and self.load_balancer_mode:
+                self._clients[desired_addr] = await self._resolve_broker(desired_addr)
+            else:
+                self._clients[desired_addr] = await self.new(desired_addr)
 
-        if addr not in self._clients:
-            self._clients[addr] = await self.new(addr)
+        assert self._clients[desired_addr].is_started
+        return self._clients[desired_addr]
 
-        assert self._clients[addr].is_started
-        return self._clients[addr]
+    async def _resolve_broker(self, addr: Addr) -> Client:
+        desired_host, desired_port = addr[0], str(addr[1])
+
+        connection_attempts = 0
+
+        while connection_attempts < self.max_retries:
+            client = await self.new(self.addr)
+
+            if client.server_properties is not None:
```

Comment:
I'd rather remove this `if` statement because we're not expecting `client.server_properties` to be None here. Instead we could add `assert client.server_properties is not None` so we prevent possible error hiding and the typechecker should be fine with that